### ISSUE TITLE
Develop advanced `docker run -a ...` tests

### ISSUE THREAD

**Issue by ldoktor** [autotest/autotest-docker#217]
Develop advanced `docker run -a ...` test, which focuses on different types of executed commands:
1. execute the test
2. (interactive only) execute command using STDIN
3. interrupt execution (ctrl+c)

variants:
1. interactive mode (eg. `bash`)
2. infinite command (eg. `ping 127.0.0.1`)
3. infinite script (eg. `ping 127.0.0.1; true`)

variants:
1. tty
2. no-tty

variants:
1. `docker run`
2. `docker attach` (run with -a ... and than attach and test on the attached process)

variants:
1. without -a
2. -a stdin
3. -a stdout
4. -a stderr
5. -a stdin -a stdout
6. -a stdin -a stderr
7. -a stdout -a stderr
8. -a stdin -a stdout -a stderr
9. random order combination (eg. -a stdout -a stdin)

It would be nice to have `cartesian_config`-like support in `docker-test` to combine the variants, otherwise we'd need to write 108 subsubtests...

**Comment by cevich** [comment]
Moving to 0.8.4 since this depends on #436
